crates/sui-framework/packages/dwallet-system/sources/dwallet.move

```diff
@@ -0,0 +1,265 @@
+        message_approval.message
+    }
+
+    public fun remove_message_approval(message_approval: MessageApproval): (ID, vector<u8>) {
+        let MessageApproval {
+            dwallet_cap_id,
+            message
+        } = message_approval;
+        (dwallet_cap_id, message)
+    }
+
+    public fun create_shared_partial_user_signed_messages<S: store, E: store + copy + drop>(
+        partial_user_signed_messages: PartialUserSignedMessages<S, E>,
+        ctx: &mut TxContext
+    ) {
+        let holder = SharedPartialUserSignedMessages {
+            id: object::new(ctx),
+            partial_user_signed_messages,
+        };
+        transfer::share_object(holder);
+    }
+
+    public fun sign_shared<S: store, E: store + copy + drop>(
+        shared: SharedPartialUserSignedMessages<S, E>,
+        message_approvals: vector<MessageApproval>,
+        ctx: &mut TxContext
+    ) {
+        let SharedPartialUserSignedMessages {
+            id,
+            partial_user_signed_messages,
+        } = shared;
+        object::delete(id);
+        sign(partial_user_signed_messages, message_approvals, ctx)
+    }
+
+    public(friend) fun create_partial_user_signed_messages<S: store, E: store + copy + drop>(
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        messages: vector<vector<u8>>,
+        sign_data: S,
+        sign_data_event: E,
+        ctx: &mut TxContext
+    ): PartialUserSignedMessages<S, E> {
+        PartialUserSignedMessages {
+            id: object::new(ctx),
+            dwallet_id,
+            dwallet_cap_id,
+            messages,
+            sign_data,
+            sign_data_event,
+        }
+    }
+
+    public fun partial_user_signed_messages_dwallet_id<S: store, E: store + copy + drop>(
+        partial_user_signed_messages: &PartialUserSignedMessages<S, E>
+    ): ID {
+        partial_user_signed_messages.dwallet_id
+    }
+
+    public fun partial_user_signed_messages_dwallet_cap_id<S: store, E: store + copy + drop>(
+        partial_user_signed_messages: &PartialUserSignedMessages<S, E>
+    ): ID {
+        partial_user_signed_messages.dwallet_cap_id
+    }
+
+    public fun partial_user_signed_messages_messages<S: store, E: store + copy + drop>(
+        partial_user_signed_messages: &PartialUserSignedMessages<S, E>
+    ): vector<vector<u8>> {
+        partial_user_signed_messages.messages
+    }
+
+    public fun sign<S: store, E: store + copy + drop>(
+        partial_user_signed_messages: PartialUserSignedMessages<S, E>,
+        message_approvals: vector<MessageApproval>,
+        ctx: &mut TxContext
+    ) {
+        let PartialUserSignedMessages {
+            id,
+            dwallet_id,
+            dwallet_cap_id,
+            messages,
+            sign_data,
+            sign_data_event,
+        } = partial_user_signed_messages;
+
+        object::delete(id);
+        let i = 0;
+        let messages_len = vector::length(&messages);
```

Comment:
When doing integer arithmetics, it is much clearer and safe to explicitly note the types. Is it possible to do so in move?
E.g. `let messages_len: u64 = vector::length(&messages);`
It is important to see types are of the same or compatible size and sign, e.g. for making sure no overflow occurrs